heidrun/uva_mods.rb

```diff
@@ -0,0 +1,182 @@
+                    .field('mods:name', 'mods:namePart'))
+
+  subjects.concat(record['mods:name']
+                    .match_attribute(:name, 'personal')
+                    .match_attribute(:authority, 'lcnaf')
+                    .field('mods:name', 'mods:namePart'))
+end
+
+Krikri::Mapper.define(:uva_mods,
+                      parser: Krikri::ModsParser) do
+  # edm:dataProvider
+  #   <location><physicalLocation>
+  dataProvider class: DPLA::MAP::Agent do
+    label record.field('mods:location', 'mods:physicalLocation')
+  end
+
+  # edm:isShownAt
+  #   <location><url access="object in context">
+  isShownAt class: DPLA::MAP::WebResource do
+    uri record.field('mods:location', 'mods:url')
+              .match_attribute(:access, 'object in context')
+  end
+
+  # edm:preview
+  #   <url access="preview">
+  preview class: DPLA::MAP::WebResource do
+    uri record.field('mods:location', 'mods:url')
+              .match_attribute(:access, 'preview')
+  end
+
+  # edm:provider
+  #   University of Virginia Library
+  provider class: DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/virginia'
+    label 'University of Virginia Library'
+  end
+
+  # dpla:originalRecord
+  #   DPLA
+  originalRecord class: DPLA::MAP::WebResource do
+    uri record_uri
+  end
+
+  # edm:hasView
+  #   <PhysicalDescription><internetMediaType>
+  hasView class: DPLA::MAP::WebResource do
+    dcformat record.field('mods:physicalDescription', 'mods:internetMediaType')
+  end
+
+  # dpla:SourceResource
+  sourceResource class: DPLA::MAP::SourceResource do
+    # dcterms:isPartOf
+    #   <relatedItem type="series" ...><titleInfo><title>[IGNORE <nonSort>]
+    collection class: DPLA::MAP::Collection,
+               each: record.field('mods:relatedItem')
+                              .match_attribute(:type, 'series')
+                              .field('mods:titleInfo', 'mods:title'),
+               as: :collection do
+      title collection
+    end
+    # <nonSort> is a child of <titleInfo> so quietly ignored - JB
+
+    # dcterms:creator
+    #   <name type="personal" authority="lcnaf">
+    #   or <name type="corporate" authority="lcnaf">
+    creator class: DPLA::MAP::Agent,
+            each: record.field('mods:name')
+                        .match_attribute(:authority, 'lcnaf')
+                        .match_attribute(:type) { |type|
+                          %w(personal corporate).include?(type)
+                        },
+            as: :creator do
+      providedLabel creator
+    end
+
+    # dc:date
+    #   <dateIssued keyDate="yes">[value] Date</dateIssued>
+    #   NOTE: if value is "unknown," do not display
+    date class: DPLA::MAP::TimeSpan,
+         each: record.field('mods:originInfo', 'mods:dateCreated')
+                     .match_attribute(:keyDate, 'yes'),
+         as: :date do
+      providedLabel date
+    end
+    # TODO: <dateIssued> is in collection-description-mods.xml
```

Comment:
Per the old system ([see previous code](https://github.com/dpla/ingestion/blob/develop/lib/mappers/uva_mapper.py#L104-L121)), we should use `dateIssued` first if it exists, and then use `dateCreated`. cc @guegueng in case she has any additional feedback